src/python_testing/matter_testing_support.py

```diff
@@ -1305,11 +1305,20 @@ def wait_for_user_input(self,
         Returns:
             str: User input or none if input is closed.
         """
+
+        # TODO(#31928): Remove any assumptions of test params for endpoint ID.
+
+        # Get the endpoint user param instead of `--endpoint-id` result, if available, temporarily.
+        endpoint_id = self.user_params.get("endpoint", None)
+        if endpoint_id is None or not isinstance(endpoint_id, int):
+            endpoint_id = self.matter_test_config.endpoint
+
         if self.runner_hook:
             self.runner_hook.show_prompt(msg=prompt_msg,
                                          placeholder=prompt_msg_placeholder,
-                                         default_value=default_value)
-        logging.info("========= USER PROMPT =========")
+                                         default_value=default_value,
```

Comment:
The hooks object for the test harness won't actually do anything with endpoint currently, so you're probably better off to bake this into the prompt message.

Reply:
Make sense @cecille, from TH perspective we won't handle endpoint argument, so it may not be necessary to be included as argument at show prompt hook method.